test/FunctionalTests/Microsoft.OData.Core.Tests/IntegrationTests/Evaluation/AutoComputePayloadMetadataInJsonIntegrationTests.cs

```diff
@@ -1356,6 +1375,172 @@ private static IEdmModel GetModelWithAbstractType(out EdmEntitySet entitySet)
+            {
+                ServiceRoot = new Uri("http://example.com"),
+                SelectAndExpand = result
+            };
+
+            Uri requestUri = new Uri("http://example.com/EntitySet(1)");
+            odataUri.RequestUri = requestUri;
+            odataUri.Path = new ODataUriParser(Model, new Uri("http://example.com"), requestUri).ParsePath();
+            settings.ODataUri = odataUri;
+
+            string output;
+            using (var messageWriter = new ODataMessageWriter(message, settings, Model))
+            {
+                ODataWriter writer = messageWriter.CreateODataResourceWriter(EntitySet, EntityType);
+
+                ODataResource resource = new ODataResource
+                {
+                    TypeName = "Namespace.EntityType",
+                    Properties = new[]
+                    {
+                        new ODataProperty { Name = "Name", Value = "SampleName" }
+                    }
+                };
+
+                if (metadataLevel == "full")
+                {
+                    resource.Id = new Uri("http://example.com/EntitySet(1)");
+                }
+
+                writer.WriteStart(resource);
+                ODataNestedResourceInfo nestedResourceInfo = new ODataNestedResourceInfo
+                {
+                    IsCollection = true,
+                    Name = "ContainedNavProp"
+                };
+                writer.WriteStart(nestedResourceInfo);
+                writer.WriteStart(new ODataResourceSet());
+                writer.WriteEnd(); // end of nested resource set
+                writer.WriteEnd(); // end of nested resource info
+                writer.WriteEnd(); // end of resource
+
+                outputStream.Seek(0, SeekOrigin.Begin);
+                output = new StreamReader(outputStream).ReadToEnd();
+            }
+
+            if (metadataLevel == "full")
+            {
+                Assert.Equal("{\"@odata.context\":\"http://example.com/$metadata#EntitySet(Name,ContainedNavProp())/$entity\"," +
+                  "\"@odata.type\":\"#Namespace.EntityType\"," +
+                  "\"@odata.id\":\"http://example.com/EntitySet(1)\"," +
+                  "\"@odata.etag\":\"W/\\\"'SampleName'\\\"\"," +
+                  "\"@odata.editLink\":\"EntitySet(1)\"," +
+                  "\"@odata.mediaEditLink\":\"EntitySet(1)/$value\"," +
+                  "\"Name\":\"SampleName\"," +
+                  "\"ContainedNavProp@odata.associationLink\":\"http://example.com/EntitySet(1)/ContainedNavProp/$ref\"," +
+                  "\"ContainedNavProp@odata.navigationLink\":\"http://example.com/EntitySet(1)/ContainedNavProp\"," +
+                  "\"ContainedNavProp\":[]}", output);
+            }
+            else
+            {
+                Assert.Equal("{\"@odata.context\":\"http://example.com/$metadata#EntitySet(Name,ContainedNavProp())/$entity\"," +
+                  "\"Name\":\"SampleName\"," +

```

Comment:
this is the scenario that (prior to this PR) threw the exception, right? 

What would the behavior be if we *did* include the key properties?

Reply:
I added a new test case to cover this.